tokio/src/sync/semaphore.rs

```diff
@@ -176,6 +298,32 @@ impl Semaphore {
     /// Otherwise, this returns a [`OwnedSemaphorePermit`] representing the
     /// acquired permit.
     ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::sync::Arc;
+    /// use tokio::sync::Semaphore;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let semaphore = Arc::new(Semaphore::new(2));
+    ///     let sem_clone = semaphore.clone();
+    ///     let handle = tokio::spawn(async move {
+    ///         let permit = sem_clone.acquire_owned().await;
+    ///         assert_eq!(permit.is_ok(), true);
+    ///     });
+    ///     {
+    ///         let p1 = semaphore.clone().acquire_owned().await;
+    ///         assert_eq!(semaphore.available_permits(), 1);
+    ///         let p2 = semaphore.clone().acquire_owned().await;
+    ///         assert_eq!(semaphore.available_permits(), 0);
+    ///         assert_eq!(p1.is_ok(), true);
+    ///         assert_eq!(p2.is_ok(), true);
+    ///     } // release the permits so the task can acquire them
+    ///     handle.await.unwrap();
```

Comment:
The primary use case of the `_owned` methods is that the permit can moved from one task to another after acquiring it.  I think this example could just be a copy of the for loop example from the beginning.